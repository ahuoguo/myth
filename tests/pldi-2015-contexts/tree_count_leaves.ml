#use "context_bool.decls"

let tree_count_leaves : tree -> nat |>
  { Leaf => 1
  | Node (Leaf, True, Leaf) => 2
  | Node (Node (Leaf, True, Leaf), True, Leaf) => 3
  | Node (Leaf, True, Node (Leaf, True, Leaf)) => 3
  | Node (Node (Node (Leaf, True, Leaf), True, Leaf), True, Leaf) => 4
  | Node (Node (Leaf, True, Leaf), True, Node (Leaf, True, Leaf)) => 4
  | Node (Node (Leaf, True, Leaf), True, Node (Node (Leaf, True, Leaf), True, Node (Leaf, True, Leaf))) => 6
  } = ?
