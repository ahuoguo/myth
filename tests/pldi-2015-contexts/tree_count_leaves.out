(* Generated Thu Nov 13 04:41:52 2014 *)
let tree_count_leaves : tree -> nat =
  let rec f1 (t1:tree) : nat =
    match t1 with
      | Leaf -> S (O)
      | Node (t2, b1, t3) -> plus (f1 t2) (f1 t3)
  in
    f1
;;
